### ISSUE TITLE
Perplexity and Accuracy scores

### ISSUE THREAD

**Issue by sukannyapurkayastha** [xiaoyuin/tntspa#2]
Hi, 
   How to find the perplexity and accuracy scores for the datasets? Is there a readme I can follow to generate those scores from your codebase?

**Comment by xiaoyuin** [comment]
Hi, 

I am not sure if I understood your question correctly. There are no individual "scores" attached to each dataset, you have to train models to get them. 

If you were just looking for results of my training on the datasets, you can access them in the folder `/results_raw` and `/results`, where I saved the output translations and training logs. If you wanted to train the models yourself and find the scores. For `nmt` framework, there's a visualization tool called [TensorBoard](https://www.tensorflow.org/tensorboard) you can use to monitor your training statistics. For `fairseq`, I was not aware of such tool so I created a python script [plot.py](https://github.com/xiaoyuin/tntspa/blob/master/plot.py) myself to read the training logs and generate the loss graph.

Regarding README, unfortunately I won't be able to write one now. The scripts saved in this repository were mainly targeted to be run on an HPC server so it will be quite complicated to describe how to run everything on a local PC.

Hope it helps

**Comment by sukannyapurkayastha** [comment]
Hi, 
   I have trained models using fairseq but fairseq only provides the BLEU scores. How did you find the accuracy as well as the F1 score for the entire dataset? Also, how to find the perplexity since fairseq provides log-likelihood scores for each query. Please let me know the formula used to generate the scores for the entire dataset.  Thanks.

**Comment by xiaoyuin** [comment]
Oh Really, interesting, then it seems fairseq has changed since I used it, because from what I checked in my training logs, fairseq provided perplexity and loss but except bleu scores during training (e.g. [my training of ConvS2S on monument dataset](https://github.com/xiaoyuin/tntspa/blob/master/results_raw/monument_600/run1/fconv_wmt_en_de/train_fairseq_fconv-3271974.out), 'ppl'=perplexity). For Bleu and accuracy, I had to write an extra script to parse the output translations and calculate myself. (pls refer to [this file](https://github.com/xiaoyuin/tntspa/blob/master/generate.py)) I don't remember using F1 score anywhere, but I suppose you can easily find the formula on the Internet. The only thing that matters is you have obtained all the output translations from the framework, based on that you can always write the analysis script on your own rather than rely on the framework's functionality.

This is the script I used to train the models with fairseq: [link](https://github.com/xiaoyuin/tntspa/blob/master/scripts/hpc_train_gpu1_fairseq.sh) (do not run it, it's for HPC). You can use it as a reference to compare what's the difference between your training commands and mine.

Anyway, I suggest you to try writing your own scripts instead of reusing mine, which I didn't put too much effort on readability so it's probably gonna spend you longer to understand. And the version of fairseq and nmt I used has been out of date for a long time, so the new output format you got could change and very likely will lead to failure of my old scripts.
